lib/config.js

```diff
@@ -127,7 +127,7 @@ var ENV_TABLE = {
   metricsInterval: 'ELASTIC_APM_METRICS_INTERVAL',
   metricsLimit: 'ELASTIC_APM_METRICS_LIMIT',
   payloadLogFile: 'ELASTIC_APM_PAYLOAD_LOG_FILE',
-  sanitizeFieldNames: 'ELASTIC_SANITIZE_FIELD_NAMES',
+  sanitizeFieldNames: ['ELASTIC_APM_SANITIZE_FIELD_NAMES', 'ELASTIC_SANITIZE_FIELD_NAMES'],
```

Comment:
As mentioned at https://github.com/elastic/apm-agent-nodejs/pull/2637#discussion_r842154360 the legacy ones should go first so that the `ELASTIC_APM_...`-prefix ones win if both are specified.

For example, currently `ELASTIC_SANITIZE_FIELD_NAMES` will win:

```
% cat foo.js
var apm = require('./').start()
console.log('sanitizeFieldNames:', apm._conf.sanitizeFieldNames)

% ELASTIC_APM_SANITIZE_FIELD_NAMES=correct ELASTIC_SANITIZE_FIELD_NAMES=deprecated node foo.js
sanitizeFieldNames: [ 'deprecated' ]
```

Reply:
57b2c70824e1ddd75353d4721fa0125f1e5fef5e